api/v1alpha1/test/new.go

```diff
@@ -94,10 +94,12 @@ func NewExtendedDaemonSetReplicaSet(ns, name string, options *NewExtendedDaemonS
 			APIVersion: apiVersion,
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace:   ns,
-			Name:        name,
-			Labels:      map[string]string{},
-			Annotations: map[string]string{},
+			Namespace: ns,
+			Name:      name,
+			Labels:    map[string]string{},
+			Annotations: map[string]string{
+				datadoghqv1alpha1.ExtendedDaemonSetReplicaSetUnreadyPodsAnnotationKey: "0",
```

Comment:
I think it is better to keep the `Annotations` by default empty, but us the `NewExtendedDaemonSetReplicaSetOptions` to add the annotations in some test cases.

I will be better to validate that the logic still work even if the annotation is not present.

Reply:
Makes sense.

I updated the code. I had to default the number of unready pods to 0 in case the annotation didn't exist.